frontend/src/lib/components/InsightLegend/InsightLegend.tsx

```diff
@@ -0,0 +1,69 @@
+import './InsightLegend.scss'
+import React from 'react'
+import { Button, Row, Col } from 'antd'
+import clsx from 'clsx'
+import { useActions, useValues } from 'kea'
+import { LegendIcon } from 'lib/components/icons'
+import { insightLogic } from 'scenes/insights/insightLogic'
+import { trendsLogic } from 'scenes/trends/trendsLogic'
+import { InsightLabel } from 'lib/components/InsightLabel'
+import { getChartColors } from 'lib/colors'
+import { PHCheckbox } from 'lib/components/PHCheckbox'
+import { formatCompareLabel } from 'scenes/insights/InsightsTable/InsightsTable'
+
+export function InsightLegendButton(): JSX.Element {
+    const { filters } = useValues(insightLogic)
+    const { toggleInsightLegend } = useActions(insightLogic)
+
+    return (
+        <Button className="insight-legend-button" onClick={toggleInsightLegend}>
+            <LegendIcon />
+            <span className="insight-legend-button-title">{filters.legend_hidden ? 'Show' : 'Hide'} Legend</span>
+        </Button>
+    )
+}
+
+export function InsightLegend(): JSX.Element {
+    const { insightProps, filters } = useValues(insightLogic)
+    const logic = trendsLogic(insightProps)
+    const { indexedResults, visibilityMap } = useValues(logic)
+    const { toggleVisibility } = useActions(logic)
+    const colorList = getChartColors('white', indexedResults.length, !!filters.compare)
+
+    return (
+        <div className={clsx('insight-legend-menu', indexedResults.length <= 5 && 'short')}>
+            {indexedResults &&
+                indexedResults.map((item) => {
+                    return (
+                        <Row key={item.id} className="insight-legend-menu-item" wrap={false}>
+                            <Col>
+                                <PHCheckbox
+                                    color={colorList[item.id]}
+                                    checked={visibilityMap[item.id]}
+                                    showIcon={indexedResults.length > 1}
```

Comment:
Feels kind of like we're abusing the PHCheckbox by disbaling it and removing the checkbox. Should we just draw a box instead of adding the `showIcon` prop?

Reply:
On second thought - it doesn't feel too weird to just leave the checkbox functionality even if there is only one item. Maybe that's the right thing here?

Reply:
The reason I changed it to hide the check was because I found myself clicking the checkbox trying to toggle it on and off. Maybe it makes sense not to disabled this on single series data altogether? It doesn't make sense that the user shouldn't have to power to hide everything if they wanted to?

Reply:
Yeah, not disabling on a single series makes sense to me.